### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by justinfagnani**
Just a minimal port, mainly to get an ES5 build for node 0.12.x support.

### CODE REVIEW

**Comment by justinfagnani** [comment]
cc @garlicnation

File: custom_typings/find-port.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare module 'find-port' {
+  module findPort {
+    function findPort();
+  }
+
+  export = findPort.findPort;
```
Comment (rictic) [inline custom_typings/find-port.d.ts]: Try this:

```typescript
declare module 'find-port' {
  module findPort {
  }
  function findPort(): number;
  export = findPort;
}
```

File: custom_typings/send.d.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+declare module 'send' {
+  module send {
+    function send();
+  }
+
+  export = send.send;
```
Comment (rictic) [inline custom_typings/send.d.ts]: Ditto

File: src/make_app.ts [hunk 3]
```diff
@@ -8,14 +8,24 @@
...
+import * as fs from 'fs';
+import * as path from 'path';
+import { parse as parseUrl } from 'url';
+import { bowerConfig } from './bower_config';
 
-const express = require('express');
-const fs = require('fs');
-const path = require('path');
-const parseUrl = require('url').parse;
 const send = require('send');
```
Comment (rictic) [inline src/make_app.ts]: Should be able to use an ES6 import with the above suggestion

File: src/start_server.ts [hunk 4]
```diff
@@ -10,11 +10,12 @@
 
 'use strict';
 
-const express = require('express');
+import * as express from 'express';
+import * as http from 'http';
+import { makeApp } from './make_app';
+import * as opn from 'opn';
+
 const findPort = require('find-port');
```
Comment (rictic) [inline src/start_server.ts]: ES6 import should be possible now

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
+    "compilerOptions": {
+        "target": "es5",
+        "module": "commonjs",
+        "moduleResolution": "node",
+        "isolatedModules": false,
+        "experimentalDecorators": true,
+        "emitDecoratorMetadata": true,
+        "declaration": false,
+        "noImplicitAny": false,
```
Comment (rictic) [inline tsconfig.json]: We want to eventually turn this on I think

File: src/start_server.ts (hunk 4 above)
Comment (justinfagnani) [inline src/start_server.ts]: not working... that style of export isn't a valid ES6 module, are you sure it's possible?

Comment (justinfagnani) [inline src/start_server.ts]: never mind!

File: src/make_app.ts (hunk 3 above)
Comment (justinfagnani) [inline src/make_app.ts]: done

File: custom_typings/send.d.ts (hunk 2 above)
Comment (justinfagnani) [inline custom_typings/send.d.ts]: done

File: custom_typings/find-port.d.ts (hunk 1 above)
Comment (justinfagnani) [inline custom_typings/find-port.d.ts]: done

**Comment by rictic** [comment]
LGTM if we put a TODO somewhere to eventually turn on noImplicitAny

File: tsconfig.json (hunk 5 above)
Comment (justinfagnani) [inline tsconfig.json]: turned on and fixed errors.

**Pull request merged**